fitlins/interfaces/afni.py

```diff
@@ -0,0 +1,624 @@
+            )
+
+            # Get boolean to index appropriate values
+            stat_bool = stat_types == contrast_type.upper()
+            contrast_bool = np.array([((name in x) and (('stim_' + name ) not in x)) for x in vol_labels])
+
+            # Get index for stdev
+            variance_img_labels = parse_afni_ext(maps["effect_variance"])[
+                "BRICK_LABS"
+            ].split("~")
+            stdev_bool = np.array(["StDev" == x for x in variance_img_labels])
+
+            # Indices for multi image nibabel object  should have length 1 and be integers
+            stat_idx = np.where(contrast_bool & stat_bool)[0]
+            stdev_idx = np.where(stdev_bool)[0]
+            # For multirow ftests there will be more than one index
+            effect_idx = np.where(contrast_bool & effect_bool)[0]
+
+            # Append maps:
+            # for each index into the result objects stored in maps, apply a
+            # modifying function as required and then append it to the
+            # appropriate output list type represented by map_list and write
+            # map_list to disk
+            for map_type, map_list, idx_list, mod_func in (
+                ("effect_size", effect_maps, effect_idx, None),
+                ("effect_variance", variance_maps, stdev_idx, np.square),
+                ("z_score", zscore_maps, stat_idx, None),
+                ("p_value", pvalue_maps, stat_idx, None),
+                ("stat", stat_maps, stat_idx, None),
+            ):
+
+                if len(effect_idx) > 1:
+                    continue
+                # If not defined mod_func will be identity
+                if not mod_func:
+
+                    def mod_func(x):
+                        return x
+
+                # Extract maps and info from bucket and append to relevant
+                # list of maps
+                for idx in idx_list:
+                    imgs = maps[map_type]
+                    fname = fname_fmt(name, map_type)
+                    extract_volume(imgs, idx, f"{map_type} of contrast {name}", fname_fmt(name, map_type))                    
+                    map_list.append(fname)
+
+        self._results["effect_maps"] = effect_maps
+        self._results["variance_maps"] = variance_maps
+        self._results["stat_maps"] = stat_maps
+        self._results["zscore_maps"] = zscore_maps
+        self._results["pvalue_maps"] = pvalue_maps
+        self._results["contrast_metadata"] = contrast_metadata
+
+    def get_stim_labels(self):
+        # Iterate through all weight specifications to get a list of stimulus
+        # column labels.
+        weights = _flatten([x["weights"] for x in self.inputs.contrast_info])
+        return list(set(_flatten([x.keys() for x in weights])))
+
+def extract_volume(imgs, idx, intent_name, fname):
+    img = imgs.slicer[..., int(idx)]
+    intent_info = get_afni_intent_info_for_subvol(img, idx)
```

Comment:
In theory, `slicer` should also slice the header, so a future-safe way to do this would be to just work from the series.

```suggestion
    intent_info = get_afni_intent_info_for_subvol(imgs, idx)
```

Reply:
The image series are atypical in that header information expected to be similar across nii volumes  are different (using AFNI buckets to prevent several files being written to disk). Since buckets aren't going to change (sensible reduction in output file count), and the nifti standard isn't going to change, this situation seems like an edge case that will persist. So yes, get_afni_intent_info_for_subvol should run on the full series and slicer could be reasonably expected to slice the header.  I'll change get_afni_intent_info accordingly